polkadot/xcm/src/v4/mod.rs

```diff
@@ -1317,8 +1318,13 @@ impl<Call> TryFrom<NewInstruction<Call>> for Instruction<Call> {
 			HrmpChannelAccepted { recipient } => Self::HrmpChannelAccepted { recipient },
 			HrmpChannelClosing { initiator, sender, recipient } =>
 				Self::HrmpChannelClosing { initiator, sender, recipient },
-			Transact { origin_kind, require_weight_at_most, call } =>
-				Self::Transact { origin_kind, require_weight_at_most, call: call.into() },
+			Transact { origin_kind, mut call } => {
+				// TODO: This requires a FRAME dependency on the core XCM crate.
+				// I wouldn't want to have to do that, but it allows for better conversions
+				// from v5 to v4.
```

Comment:
In crates like `xcm-executor` and `xcm-builder` we expect `RuntimeCall`s that implement `GetDispatchInfo` already. But in the `xcm` crate itself we didn't have this FRAME dependency before. What do you think @acatangiu ?

Reply:
After thinking about it for a while, I removed that conversion and just used a large enough weight. We don't use conversions from latest to the previous one that much, it's usually the other way around.
That said, I can change it if we think it's worth it to guarantee the conversion works.

Reply:
I think the dependency is fine and the correct conversion is totally worth it. Since `xcm-executor` already depends on FRAME this will definitely not be a problem for anyone.

Somewhere in the future we can remove it when we remove v4 (but I don't expect anyone caring).

Reply:
I'll go back then